corehq/blobs/models.py

```diff
@@ -1,22 +1,104 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
-from django.db import models
+from datetime import datetime
+from uuid import uuid4
 
+from django.db.models import (
+    BooleanField,
+    CharField,
+    DateTimeField,
+    IntegerField,
+    Model,
+    PositiveIntegerField,
+    PositiveSmallIntegerField,
+)
+from partial_index import PartialIndex
 
-class BlobMigrationState(models.Model):
-    slug = models.CharField(max_length=20, unique=True)
-    timestamp = models.DateTimeField(auto_now=True)
+from corehq.sql_db.models import PartitionedModel, RestrictedManager
 
+from .util import NullJsonField
 
-class BlobExpiration(models.Model):
+
+def uuid4_hex():
+    return uuid4().hex
+
+
+class BlobMeta(PartitionedModel, Model):
+    """Metadata about an object stored in the blob db"""
+
+    partition_attr = "parent_id"
+    objects = RestrictedManager()
+
+    domain = CharField(max_length=255)
+    parent_id = CharField(
+        max_length=255,
+        help_text="Parent primary key or unique identifier",
+    )
+    name = CharField(
+        max_length=255,
+        default="",
+        help_text="""Optional blob name.
+
+        This field is intended to be used by doc types having multiple
+        blobs associated with a single document.
+        """,
+    )
+    key = CharField(
+        max_length=255,
```

Comment:
does this need to be 255?

Reply:
probably not, but other models use 255

https://github.com/dimagi/commcare-hq/blob/5ba3174d388d369ea0545beb7c06e4eb02b29ad2/corehq/form_processor/models.py#L424

I'm pretty sure VARCHAR and TEXT fields only use as much space as the data stored in them, and the length is only an upper limit, so I'm not sure it matters that much. Historical blob key lengths for blobs created before the new API is in use are bucket length + identifier length, which is typically len(couch_dbname) + len(couch_id) + len(identifier) + 2. Going forward, it should be the length of a UUID hex string (32 characters), which seems much more reasonable.